pkg/integrations/v2/register.go

```diff
@@ -1,85 +1,5 @@
 package integrations
 
-import (
-	"fmt"
-	"reflect"
-	"strings"
-	"testing"
-
-	"gopkg.in/yaml.v2"
-
-	v1 "github.com/grafana/agent/pkg/integrations"
-	"github.com/grafana/agent/pkg/integrations/v2/common"
-	"github.com/grafana/agent/pkg/util"
-)
-
-var (
-	integrationNames = make(map[string]interface{})  // Cache of field names for uniqueness checking.
-	configFieldNames = make(map[reflect.Type]string) // Map of registered type to field name
-	integrationTypes = make(map[reflect.Type]Type)   // Map of registered type to Type
-
-	// Registered integrations. Registered integrations may be either a Config or
-	// a v1.Config. v1.Configs must have a corresponding upgrader for their type.
-	registered = []interface{}{}
-	upgraders  = make(map[string]UpgradeFunc)
-
-	emptyStructType = reflect.TypeOf(struct{}{})
-	configsType     = reflect.TypeOf(Configs{})
-)
-
-// Register dynamically registers a new integration. The Config
-// will represent the configuration that controls the specific integration.
-// Registered Configs may be loaded using UnmarshalYAML or manually
-// constructed.
-//
-// ty controls how the integration can be unmarshaled from YAML.
-//
-// Register panics if cfg is not a pointer.
-func Register(cfg Config, ty Type) {
-	registerIntegration(cfg, cfg.Name(), ty, nil)
-}
-
-func registerIntegration(v interface{}, name string, ty Type, upgrader UpgradeFunc) {
-	if reflect.TypeOf(v).Kind() != reflect.Ptr {
-		panic(fmt.Sprintf("Register must be given a pointer, got %T", v))
-	}
-	if _, exist := integrationNames[name]; exist {
-		panic(fmt.Sprintf("Integration %q registered twice", name))
-	}
-	integrationNames[name] = v
-
-	registered = append(registered, v)
-
-	configTy := reflect.TypeOf(v)
-	integrationTypes[configTy] = ty
-	configFieldNames[configTy] = name
-	upgraders[name] = upgrader
-}
-
-// RegisterLegacy registers a v1.Config. upgrader will be used to upgrade it.
-// upgrader will only be invoked after unmarshaling cfg from YAML, and the
-// upgraded Config will be unwrapped again when marshaling back to YAML.
-//
-// RegisterLegacy only exists for the transition period where the v2
-// integrations subsystem is an experiment. RegisterLegacy will be removed at a
-// later date.
-func RegisterLegacy(cfg v1.Config, ty Type, upgrader UpgradeFunc) {
-	registerIntegration(cfg, cfg.Name(), ty, upgrader)
-}
-
-// UpgradeFunc upgrades cfg to a UpgradedConfig.
-type UpgradeFunc func(cfg v1.Config, common common.MetricsConfig) UpgradedConfig
-
-// UpgradedConfig is a v2 Config that was constructed through a legacy
-// v1.Config. It allows unwrapping to retrieve the original config for
-// the purposes of marshaling or unmarshaling.
-type UpgradedConfig interface {
-	Config
-
-	// LegacyConfig returns the old v1.Config.
-	LegacyConfig() (v1.Config, common.MetricsConfig)
-}
```

Comment:
Should these be kept until we know we definitely don't need them later on?

Reply:
If needed later, then I can re-add. If they are needed then they will likely take on a slightly different form.